src/app/shared/table-footer/table-footer.component.ts

```diff
@@ -0,0 +1,66 @@
+import { Component, Input, OnInit, Output, EventEmitter, OnChanges } from '@angular/core';
+
+@Component({
+	selector: 'table-footer',
+	templateUrl: './table-footer.component.html',
+	styleUrls: ['./table-footer.component.scss']
+})
+export class TableFooterComponent implements OnInit {
+	private _currentPage: number = 1;
+	private _itemsPerPage: number;
+	public nrOfPages: number;
+
+	@Input() set itemsPerPage(value) {
+		this._itemsPerPage = value;
+		this.getNumberOfPages();
+	}
+	@Input() nrOfItems: number;
+	@Output() page = new EventEmitter();
+
+	ngOnInit() {
+		this.getNumberOfPages();
+	}
+
+	private getNumberOfPages() {
+		this.nrOfItems = Number(this.nrOfItems);
+		this._itemsPerPage = Number(this._itemsPerPage);
```

Comment:
you can use +nrOfItems to convert to number, instead. also, you shouldn't convert something to number if you expect it to be a number by default, rather convert it wherever you don't know for sure it's type, like when reading from the select you mentioned